Des58 light dark mode

AndrewLin95 opened the pull request:
Updated the project to support Dark and Light mode.

The main changes were the addition of the `DarkMode.tsx` and `providers.tsx` files under the components > DarkMode folder. I added a new package `next-themes` which provides a useTheme hook. This hook allows us to modify the HTML class allowing tailwind to pickup on the changes to alter the theme7  from dark to light. The provider wraps the `layout.tsx` file allowing the `DarkMode.tsx` file to be used throughout the project.

![image](https://github.com/AndrewLin95/next_destination_tracker/assets/107276941/91a8735e-8e8e-4399-828f-e25f23258dda)

`tailwind.config.js` was modified to include the updated color palette. Refer to the specific file for more information.
![image](https://github.com/AndrewLin95/next_destination_tracker/assets/107276941/838db161-fe44-4266-8e52-ca60d1de1aa2)

TODO in a later PR:
 - To refactor the reusable styling code to a separate `stylingConstants` file so styles are more reusable, easier to maintain and more consistent.

skim2264 reviewed (changes requested):
The light and dark modes look great and the toggle functions work well! Just 4 little things I noticed:

1. The filter dropdown is a little hard to read, so can we change it to be more opaque so the text underneath isn't visible and collides with the filter text.
2. The text of the notes in the schedule in light mode is harder to read with the text-shadow so I think it would be best to remove the text-shadow or to modify the color.
3. The light/dark mode symbol on the button needs more contrast from the background to be more visible.
4. Some buttons in dark mode do not have the hover transition anymore when they are present in light mode (e.g. Apply filter and cancel button in filter modal, buttons on homepage for track new project + upload banner, and edit Note buttons)

AndrewLin95 commented:
> The light and dark modes look great and the toggle functions work well! Just 4 little things I noticed:
> 
> 1. The filter dropdown is a little hard to read, so can we change it to be more opaque so the text underneath isn't visible and collides with the filter text.
> 2. The text of the notes in the schedule in light mode is harder to read with the text-shadow so I think it would be best to remove the text-shadow or to modify the color.
> 3. The light/dark mode symbol on the button needs more contrast from the background to be more visible.
> 4. Some buttons in dark mode do not have the hover transition anymore when they are present in light mode (e.g. Apply filter and cancel button in filter modal, buttons on homepage for track new project + upload banner, and edit Note buttons)

1. Updated the filter dropdown to remove any opacity and added a border to further distinguish it from the text underneath.
2. I will create another Jira task for this one. The colors for the note and schedule are applied and used differently from the other parts of the code and may require a bit more work than the changes done in this PR.
3. Updated! I inverted the colors so it pops more LOL.
4. Great catch! I didn't even notice this until you pointed it out. The styling was missed from several buttons and it looks like we need a dedicated `dark:` keyword for tailwindcss to apply styling, it doesnt look like it applies these effects to dark mode by default(?).

skim2264 reviewed (approved):
It looks great!
